gobblin-runtime/src/main/java/org/apache/gobblin/runtime/TaskStateCollectorService.java

```diff
@@ -281,7 +280,8 @@ private void reportJobProgress(TaskState taskState) {
     this.bytesCopiedSoFar += taskByteSize;
     Double newPercentageCopied = this.bytesCopiedSoFar / this.totalSizeToCopy;
 
-    if (newPercentageCopied - this.lastPercentageReported > ConfigurationKeys.DEFAULT_PROGRESS_REPORTING_THRESHOLD) {
+    if (newPercentageCopied - this.lastPercentageReported >= ConfigurationKeys.DEFAULT_PROGRESS_REPORTING_THRESHOLD ||
+        (newPercentageCopied > 1 - ConfigurationKeys.DEFAULT_PROGRESS_REPORTING_THRESHOLD && Math.abs(newPercentageCopied - this.lastPercentageReported) > 0.001)) {
```

Comment:
Let's leave a comment here for what this second scenario covers. Even though it's when the job is going from 95-100%, would 0.001 be a bit low of a threshold? It'll send up to 50 events reporting from 95-100 for each job, which seems a quite different from the previous reporting thresholds.

Reply:
Yeah seems to me like we should just have a check for if the percentage has reached 100 instead of reporting more events in the last interval of the job.

Reply:
Agree with the points above. It's a choice essentially between if we want to report 96, 97, 98, 99, and 100 potentially or just report 95 and then 100. I did 0.001 as the threshold in case the 0.1% difference corresponds to the number being rounded up, but I did think it could be excessive like you are saying in the worst case. Any preferences?

Reply:
Since we have a config for progress reporting threshold set to 5%, I think it makes the most sense if that still applies even in the last interval instead of suddenly sending more events at the end of the job (other than the fix of always sending the final 100% completion event).